File: src/main/java/com/zuminX/annotations/swagger/ExtensionProperty.java
package com.zuminX.annotations.swagger;

import com.zuminX.annotations.AnnotationAttr;
import com.zuminX.annotations.AnnotationStr;
import com.zuminX.names.ClassName;
import com.zuminX.names.SwaggerAnnotation;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 对应io.swagger.annotations.ExtensionProperty
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ExtensionProperty extends AnnotationStr {

  @AnnotationAttr(show = true)
  private String name;

  @AnnotationAttr(show = true)
  private String value;

  @Override
  public ClassName getClassName() {
    return SwaggerAnnotation.EXTENSION_PROPERTY;
  }
}

File: src/main/java/com/zuminX/annotations/swagger/ExtensionProperty.kt
package com.zuminX.annotations.swagger

import com.zuminX.annotations.AnnotationAttr
import com.zuminX.annotations.AnnotationStr
import com.zuminX.names.ClassName
import com.zuminX.names.annotation.EXTENSION_PROPERTY

/**
 * 对应io.swagger.annotations.ExtensionProperty
 */
class ExtensionProperty : AnnotationStr() {

  @field:AnnotationAttr(show = true)
  var name: String? = null

  @field:AnnotationAttr(show = true)
  var value: String? = null
  
  override fun getClassName(): ClassName = EXTENSION_PROPERTY
}